nestkernel/node_collection.h

```diff
@@ -729,7 +729,7 @@ inline bool NodeCollectionPrimitive::operator==( NodeCollectionPTR rhs ) const
 {
   auto const* const rhs_ptr = dynamic_cast< NodeCollectionPrimitive const* >( rhs.get() );
 
-  return first_ == rhs_ptr->first_ and last_ == rhs_ptr->last_ and model_id_ == rhs_ptr->model_id_
+  return rhs_ptr and first_ == rhs_ptr->first_ and last_ == rhs_ptr->last_ and model_id_ == rhs_ptr->model_id_
```

Comment:
Could you insert a comment about why we need to return the pointer? Also below (though there it is the `not rhs_ptr`)

Reply:
@hakonsbm I wonder if it would not be better to be explicit here and write `rhs_ptr != 0 and ...` and correspondingly below. I think it will make the code more readable.

Reply:
@heplesser Done.